util-core/src/main/scala/com/twitter/util/BatchExecutor.scala

```diff
@@ -30,7 +31,7 @@ private[util] class BatchExecutor[In, Out](
   sizeThreshold: Int,
   timeThreshold: Duration = Duration.Top,
   sizePercentile: => Float = 1.0f,
-  f: Seq[In] => Future[Seq[Out]]
+  f: Iterable[In] => Future[Seq[Out]]
```

Comment:
This raises, rather than lowers the requirements. Still, this seems reasonable in the name of unbenchmarked performance.